include/openssl/asn1.h

```diff
@@ -141,6 +141,8 @@ DEFINE_STACK_OF(X509_ALGOR)
 # define ASN1_STRING_FLAG_MSTRING 0x040
 /* String is embedded and only content should be freed */
 # define ASN1_STRING_FLAG_EMBED 0x080
+/* String should be parsed in RFC 5280's time format */
+# define ASN1_STRING_FLAG_X509_TIME 0x100
```

Comment:
IMHO, since we need to utilize the ASN1_TIME to struct tm parsing logic, it might be one of the best ways to reduce duplicated code if we use a flag and treat it in current  ASN1_TIME to struct tm parsing functions.